### PR TITLE
feat(#1568): Added "ask ai" section session splitting function

### PR DESCRIPTION
**Pull request by GodMeowIceSun**

### CODE REVIEW

**Comment by GodMeowIceSun** [comment]
This submission mainly adds the "ask ai" session function and fine-tuning the following styles @boojack

File: web/src/components/AskAIDialog.tsx [hunk 1]
```diff
@@ -9,14 +22,19 @@ import { useMessageStore } from "@/store/zustand/message";
 import Icon from "./Icon";
 import { generateDialog } from "./Dialog";
 import showSettingDialog from "./SettingDialog";
+import { MessageGroup, useMessageGroupStore } from "@/store/zustand/message-group";
+import { PlusIcon } from "lucide-react";
+// import { useMessageGroupStore } from "@/store/zustand/message-group";
```
Comment (boojack) [inline web/src/components/AskAIDialog.tsx:27]: remove unused code

**Review by boojack — APPROVED** [review]
LGTM, otherwise

File: web/src/css/global.css [hunk 2]
```diff
@@ -9,3 +9,30 @@ body {
...
+  border-top-left-radius: 0;
+}
+
+
+.button-len-max-150 {
+  max-width: 150px; /* 按钮的最大宽度 */
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
```
[the remark this answers is not in the document]
Comment (GodMeowIceSun) [reply web/src/css/global.css:38]: Sorry, I don't know much about tailwind 😂

File: web/src/components/AskAIDialog.tsx [hunk 3]
```diff
@@ -41,7 +59,7 @@ const AskAIDialog: React.FC<Props> = (props: Props) => {
   const handleKeyDown = (event: React.KeyboardEvent) => {
     if (event.key === "Enter" && !event.shiftKey && !isInIME) {
       event.preventDefault();
-      handleSendQuestionButtonClick();
+      handleSendQuestionButtonClick().then();
```
[the remark this answers is not in the document]
Comment (GodMeowIceSun) [reply web/src/components/AskAIDialog.tsx:62]: This may not be a problem

File: web/src/components/AskAIDialog.tsx (hunk 1 above)
Comment (GodMeowIceSun) [reply web/src/components/AskAIDialog.tsx:27]: will resolve in new commit

**Comment by GodMeowIceSun** [comment]
something is new hear: 
1. Resolve the issue of components not being stretched when only user input is available
2. New session automatic switching function
3. remove unused code
4. New Remove Session Function
5. Rename some methods

@boojack

**Review by CorrectRoadH — APPROVED** [review]
LGTM

File: web/src/css/global.css (hunk 2 above)
Comment (boojack) [inline web/src/css/global.css:38]: Move them into `ask-ai-dialog.less`

**Comment by GodMeowIceSun** [comment]
i may move them in 2 days------------------------------------------------------------------
***@***.***>
日　期：2023年05月24日 20:23:21
***@***.***>
***@***.***>; ***@***.***>
主　题：Re: [usememos/memos] feat(#1568): Added "ask ai" section session splitting function (PR #1711)

@boojack commented on this pull request. 
In web/src/css/global.css:
> +.button-group {
+  display: flex;
+  gap: 0; /* 按钮之间的间距 */
+}
+
+.button-group>button:not(:first-child):not(:last-child) {
+  border-radius: 0;
+}
+
+.button-group>button:first-child {
+  border-top-right-radius: 0;
+  border-bottom-right-radius: 0;
+}
+
+.button-group>button:last-child {
+  border-bottom-left-radius: 0;
+  border-top-left-radius: 0;
+}
+
+
+.button-len-max-150 {
+  max-width: 150px; /* 按钮的最大宽度 */
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}

Move them into ask-ai-dialog.less

File: web/src/css/global.css (hunk 2 above)
Comment (boojack) [reply web/src/css/global.css:38]: I'll do it in the next PR.

**Pull request merged**